File: pentalk_front/ios/Runner/DrawingMetricsStore.swift
import CoreGraphics

enum DrawingMetricsStore {
    struct Metrics {
        let renderWidth: CGFloat
        let renderHeight: CGFloat
        let pdfWidth: CGFloat
        let pdfHeight: CGFloat
    }

    static var metrics: Metrics?

    static func update(renderWidth: CGFloat, renderHeight: CGFloat, pdfWidth: CGFloat, pdfHeight: CGFloat) {
        metrics = Metrics(
            renderWidth: renderWidth,
            renderHeight: renderHeight,
            pdfWidth: pdfWidth,
            pdfHeight: pdfHeight
        )
    }

    static func normalize(point: CGPoint) -> CGPoint {
        guard let metrics, metrics.renderWidth > 0, metrics.renderHeight > 0 else {
            return point
        }
        let nx = max(0, min(1, point.x / metrics.renderWidth))
        let ny = max(0, min(1, point.y / metrics.renderHeight))
        return CGPoint(x: nx, y: ny)
    }
}

File: pentalk_front/android/app/src/main/kotlin/com/example/pentalk_front/drawing/DrawingMetricsStore.kt
package com.example.pentalk_front.drawing

import kotlin.math.max
import kotlin.math.min

object DrawingMetricsStore {
    @Volatile
    private var renderWidth: Float = 0f
    @Volatile
    private var renderHeight: Float = 0f
    @Volatile
    private var pdfWidth: Float = 0f
    @Volatile
    private var pdfHeight: Float = 0f

    fun update(payload: Map<String, Any>) {
        renderWidth = (payload["renderWidth"] as? Number)?.toFloat() ?: 0f
        renderHeight = (payload["renderHeight"] as? Number)?.toFloat() ?: 0f
        pdfWidth = (payload["pdfWidth"] as? Number)?.toFloat() ?: 0f
        pdfHeight = (payload["pdfHeight"] as? Number)?.toFloat() ?: 0f
    }

    fun normalize(x: Float, y: Float): Pair<Float, Float> {
        if (renderWidth <= 0f || renderHeight <= 0f) {
            return Pair(x, y)
        }
        val nx = min(1f, max(0f, x / renderWidth))
        val ny = min(1f, max(0f, y / renderHeight))
        return Pair(nx, ny)
    }

    fun getPdfSize(): Pair<Float, Float> {
        return Pair(pdfWidth, pdfHeight)
    }
}
